appcontainer/AppContainerModel.ts

```diff
@@ -175,7 +175,22 @@ export class AppContainerModel extends HoistModel {
             await this.completeInitAsync();
         } catch (e) {
             this.setAppState('LOAD_FAILED');
-            XH.handleException(e, {requireReload: true});
+
+            let ex = e;
+            if (ex.isServerUnavailable) {
+                const {baseUrl} = XH,
+                    serverUrl = baseUrl.startsWith('http')
+                        ? baseUrl
+                        : `${window.location.origin}${baseUrl}`;
+
+                ex = XH.exception({
+                    name: 'Unable to Contact UI Server',
+                    message: `The web browser is unable to contact the UI server at ${serverUrl}`,
```

Comment:
How are we sure that a serverUnavailable exception is actually being thrown from the UI server?  It looks like this catch encompasses appModel.initAsync() which could presumably be making other calls.  Just checking that we should be this specific with our message and call out the UI server directly

Reply:
@amcclain suggested changes are this new PR: https://github.com/xh/hoist-react/pull/3639